modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/ManagementToolbarPropsTransformer.js

```diff
@@ -0,0 +1,252 @@
+		selectFolderURL,
+		trashEnabled,
+		viewFileEntryTypeURL,
+	},
+	portletNamespace,
+	...otherProps
+}) {
+	const processAction = (action, url) => {
+		if (!action) {
+			return;
+		}
+
+		const form = document.getElementById(`${portletNamespace}fm2`);
+
+		if (!form) {
+			return;
+		}
+
+		form.setAttribute('method', 'post');
+
+		const actionInputElement = form.querySelector(
+			`#${portletNamespace}javax-portlet-action`
+		);
+
+		if (actionInputElement) {
+			actionInputElement.setAttribute('value', action);
+		}
+
+		const commandInputElement = form.querySelector(
+			`#${portletNamespace}cmd`
+		);
+
+		if (commandInputElement) {
+			commandInputElement.setAttribute('value', action);
+		}
+
+		submitForm(form, url, false);
+	};
+
+	const checkin = () => {
+		Liferay.componentReady(
+			`${portletNamespace}DocumentLibraryCheckinModal`
+		).then((documentLibraryCheckinModal) => {
+			documentLibraryCheckinModal.open((versionIncrease, changeLog) => {
+				const form = document.getElementById(`${portletNamespace}fm2`);
+
+				if (!form) {
+					return;
+				}
+
+				const changeLogInput = form.querySelector(
+					`#${portletNamespace}changeLog`
+				);
+
+				if (changeLogInput) {
+					changeLogInput.setAttribute('value', changeLog);
+				}
+
+				const versionIncreaseInput = form.querySelector(
+					`#${portletNamespace}versionIncrease`
+				);
+
+				if (versionIncreaseInput) {
+					versionIncreaseInput.setAttribute('value', versionIncrease);
+				}
+
+				processAction('checkin', editEntryURL);
+			});
+		});
+	};
+
+	const deleteEntries = () => {
+		let action;
+
+		if (trashEnabled) {
+			action = 'move_to_trash';
+		}
+		else if (
+			confirm(
+				Liferay.Language.get(
+					'are-you-sure-you-want-to-delete-the-selected-entries'
+				)
+			)
+		) {
+			action = 'delete';
+		}
+
+		processAction(action, editEntryURL);
+	};
+
+	const editTags = () => {
+		const searchContainer = Liferay.SearchContainer.get(
+			otherProps.searchContainerId
+		);
+
+		Liferay.componentReady(`${portletNamespace}EditTagsComponent`).then(
+			(editTagsComponent) => {
+				const bulkSelection = searchContainer.select?.get(
+					'bulkSelection'
+				);
+
+				const selectedFileEntries = searchContainer.select
+					.getAllSelectedElements()
+					.get('value');
+
+				editTagsComponent.open(
+					selectedFileEntries,
+					bulkSelection,
+					folderConfiguration.defaultParentFolderId
+				);
+			}
+		);
+	};
+
+	const move = () => {
+		const searchContainer = Liferay.SearchContainer.get(
+			otherProps.searchContainerId
+		);
+
+		let selectedItems = 0;
+
+		if (searchContainer.select) {
+			selectedItems = searchContainer.select
+				.getAllSelectedElements()
+				.filter(':enabled')
+				.size();
+		}
+
+		const dialogTitle =
+			selectedItems === 1
+				? Liferay.Language.get('select-destination-folder-for-x-item')
+				: Liferay.Language.get('select-destination-folder-for-x-items');
+
+		openSelectionModal({
+			height: '480px',
```

Comment:
Should this maybe be declared as a constant at the start of the file?

Reply:
which one the height of the window?
I personally don't think so

Reply:
The `480px`, just looks like something we usually have alongside other constant definitions like `const OPEN_SELECTION_MODAL_HEIGHT = 480px`

Reply:
That might be useful when that constant is used in various places, but in this case I personally don't see any advantage.

Reply:
I'm not saying I don't want apply that change if you convince me I should it of course.

Reply:
The only reason that I can think of would be **visibility** and **ease of change** in the future. I just remember that even though some constants were used once, we would declare them at the start of the file so that if you need to change those fixed values it's easier. 
Perhaps whoever explained it to me in the past didn't go in-depth on the why, just made sure it was done, or I forgot 🤷 .
I don't care much for it, just wanted to make sure it's not something you forgot about 😄 

Here's a couple examples where it's used just once but still declared at the top: [`layout-content-page-editor-web/Topper.js`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/Topper.js#L46), [`image_selector.js`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/image_selector/js/image_selector.js#L184).

At this point I want to know if this is a pattern we want to follow or just an accident 😕 
Obligatory [song link](https://www.youtube.com/watch?v=2lvs2FzF64o) that was hyping me up while I was looking for the examples 🎸

Reply:
This would fall under subjective call of "what makes the code more readable". As far as I know, we don't have a strong convention on this code pattern. I'd say both approaches are fine and non-blocking, whatever feels better for developer on a specific case.

Reply:
Sound good @markocikos ❤️